[12.0] partner multi relation improvements

NL66278 opened the pull request:
- Black code, to enable backport/forward port fixes to/from 13.0 and beyond;
- Correct handling of inactive;
- Correct color coding of view entries;
- View fixes;
- Ensure tabs shown on all partner forms;
- Better integration into standard odoo Contacts app.

NL66278 committed:
[IMP] partner_multi_relation. More user-friendly.

NL66278 committed:
[IMP] partner_multi_relation. Tests and demo data.

Reorganize test to make more use of demo data.
Also better following of OCA standards.

NL66278 committed:
[IMP] partner_multi_relation. Easy links to underlying many2one.

- Use clickable Many2one fields in to go from relation to underlying partners and type;
- Show address complete with email in relation form.

NL66278 committed:
[FIX] partner_multi_relation. Prevent unknown fields warning.

In some cases Odoo attempts to write domain fields on res.partner.relation.all,
despite those being computed fields with store=False. Therefore clear these fields
from vals before writing underlying res.partner.relation.

NL66278 committed:
[FIX] partner_multi_relation. Put no_create in right spot.

NL66278 committed:
[FIX] partner_multi_relation. Delete obsolete test.

NL66278 committed:
[FIX] Update partner tabs for all partner actions.

NL66278 committed:
[FIX] partner_multi_relation. View fixes.

- Properly ident xml files;
- Specify mode=extension for inheriting views.

NL66278 committed:
[IMP] partner_multi_relation. Improve python headers.

NL66278 committed:
[FIX] partner_multi_relation. (De-)Archive relations.

Correctly handle active field on res.partner.relation.all.

NL66278 committed:
[IMP] parter_multi_relation(_tabs). Sanitize menu.

NL66278 committed:
[IMP] partner_multi_relation(_tabs). Black code.

Black is not required before 13.0, However to be able to easily forward port or backward port
improvements and fixes between releases, which will be done a lot for the partner_multi_relation*
modules, we do this also for the older versions.

NL66278 committed:
[FIX] partner_multi_relation(_tabs). Replace color.

color attribute on tree is deprecated since 9.0. Replace by decoration-${name}.

NL66278 committed:
[IMP] partner_multi_relation. Active search.

Proper support for search in active in res.partner.relation.all.

NL66278 committed:
[IMP] partner_multi_relation. Streamline and test.

Streamline some code in res.partner model and add tests.

patrickrwilson commented on partner_multi_relation/models/res_partner.py:
```diff
@@ -170,28 +167,26 @@ def get_partner_type(self):
-            action['context'] = context
-            return action
+        """Handle button to show relations for partner."""
+        self.ensure_one()
```
I could be wrong but isn't ensure_one deprecated?

rousseldenis replied:
Nope, ensure_one() is not removed as it can be useful when you want to ... ensure one record :-)

patrickrwilson commented on partner_multi_relation/views/res_partner.xml:
```diff
@@ -1,43 +1,53 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
These are no longer needed since v11 I believe but don't impact functionality leaving them there.

rousseldenis replied:
> These are no longer needed since v11 I believe but don't impact functionality leaving them there.

Don't misunderstand python files encoding and xml document encoding which is not equal!

patrickrwilson reviewed (approved):
Works great, especially the archive change. I did add only a few comments but no stoppers at all. Good work. LGTM.

skeller1 commented on partner_multi_relation_tabs/models/ir_actions_act_window.py:
```diff
@@ -0,0 +1,28 @@
+
+
+class IrActionsActWindow(models.Model):
+    """Make sure tabs visibility updated when loading partner forms."""
```
Brilliant, perfect solution using ir stuff for this.

Maybe you have an idea to fix the problem that the relation tab is not added to res.partner form after:

**1. Click on external link in many2one widget**
![RelationTabExternalLinkListViewRelation](https://user-images.githubusercontent.com/226753/97230361-16eef300-17da-11eb-864a-ec1da9c01c78.png)


**2. Open contact/partner using channel follower list links (chatter)**
![RelationTabOpenPartnerthorughChannelFollowerList](https://user-images.githubusercontent.com/226753/97230761-d04dc880-17da-11eb-85ce-271b08cf033f.png)


The redirect in follower.js is realized using this.do_action({})  https://github.com/odoo/odoo/blob/8325a0ec3f4ab1e7c92599789bfee5ca9bc3da6d/addons/mail/static/src/js/followers.js#L392

Maybe this is a further improvement.

Great work LGTM

NL66278 committed:
[IMP] partner_multi_relation(_tabs). Enhance relation type search.

NL66278 committed:
[IMP] partner_multi_relation_tabs. Show inactive.

Inactive connections are correctly shown on tab pages, but after active connections.

NL66278 committed:
[IMP] partner_multi_relation. Only connect contact partners.

NL66278 committed:
[FIX] partner_multi_relation. Allow export/import of relations.

NL66278 committed:
[IMP] partner_multi_relation. Remove dependency on sales_team.

Also make all configuration available to group_partner_manager.

NL66278 committed:
[FIX] partner_multi_relation_tabs. Conventions.

NL66278 commented:
To solve the problem of the tabs not adjusting in some other forms, I added improvements for both usability and code quality that I extensively tested in 10.0, and that is already used in production for two years. Unfortunately the web_domain_field module, used for this was not ported yet to OCA/web 12.0. The needed PR is here: https://github.com/OCA/web/pull/1738

NL66278 committed:
[FIX] partner_multi_relation. Show relation form.

NL66278 committed:
[FIX] partner_multi_relation. Fix mute of inactive relations

NL66278 committed:
[IMP] partner_multi_relation_tabs: apply domain field on tab

NL66278 committed:
[FIX] partner_multi_relation_tabs. Streamline.

Simpler, better, faster and also working when res.partner is not active model.

NL66278 committed:
[IMP] partner_multi_relation_tabs. Make sure tabs up to date.

Attempt to make sure tab fields created in partner are up to date when needed.

NL66278 committed:
[FIX] partner_multi_relation Show inactive connections from button.

When using the button 'Relations' in the partner form, inactive connections where not shown.

NL66278 committed:
[FIX] partner_multi_relation: Correct test for action.

NL66278 committed:
[FIX] partner_multi_relation_tabs: update tests - no more update key in context

NL66278 committed:
[FIX] partner_multi_relations: coding style

rvalyi commented:
@NL66278 is it OK for merging for you?

rvalyi commented:
@NL66278 is it OK for merging for you?

NL66278 commented:
@rvalyi Certainly, please go ahead!

rvalyi commented:
/ocabot merge major
